### PR TITLE
drivers: can: add support for Nordic nRF MCAN

### PR DESCRIPTION
**Pull request by gmarull**

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by gmarull** [comment]
last push: added `can` to the `supported` list in board yaml

File: drivers/can/can_nrf.c [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+#include <zephyr/irq.h>
+
+/* nRF CAN wrapper offsets */
+#define CAN_TASKS_START	  offsetof(NRF_CAN_Type, TASKS_START)
+#define CAN_EVENTS_CORE_0 offsetof(NRF_CAN_Type, EVENTS_CORE[0])
+#define CAN_EVENTS_CORE_1 offsetof(NRF_CAN_Type, EVENTS_CORE[1])
+#define CAN_INTEN	  offsetof(NRF_CAN_Type, INTEN)
+
+struct can_nrf_config {
+	uint32_t wrapper;
```
Comment (alexanderwachter) [inline drivers/can/can_nrf.c:26]: Just out of curiosity, what is this wrapper? Is it something related to multicore and irq?

Comment (gmarull) [reply drivers/can/can_nrf.c:26]: nope, is just the nordic-specific thin layer on top of the Bosch MCAN IP (eg IRQ enablement, events, etc.)

**Review by alexanderwachter — APPROVED** [review]
Looks good.A datasheet would be nice. When is it released?

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by carlescufi** [comment]
> Looks good.A datasheet would be nice. When is it released?

Late this year!

**Pull request merged**